src/renderer/Metal.zig

```diff
@@ -182,27 +182,46 @@ pub const GPUState = struct {
     /// This buffer is written exactly once so we can use it globally.
     instance: InstanceBuffer, // MTLBuffer
 
+    /// The default storage mode to use for resources created with our device.
+    ///
+    /// This is based on whether the device is a discrete GPU or not, since
+    /// discrete GPUs do not have unified memory and therefore do not support
+    /// the "shared" storage mode, instead we have to use the "managed" mode.
+    default_storage_mode: mtl.MTLResourceOptions.StorageMode,
+
     pub fn init() !GPUState {
         const device = try chooseDevice();
         const queue = device.msgSend(objc.Object, objc.sel("newCommandQueue"), .{});
         errdefer queue.release();
 
+        // We determine whether our device is a discrete GPU based on these:
+        // - We're on macOS (iOS, iPadOS, etc. are guaranteed to be integrated).
+        // - We're not on aarch64 (Apple Silicon, therefore integrated).
+        // - The device reports that it does not have unified memory.
+        const is_discrete =
```

Comment:
This is super minor and we don't need to address it if you don't want, but it may be helpful to make a top-level const like `const discrete_supported` with the comptime parts of this. 

I'm not sure we'd use it anywhere else but I think it'd also clean this boolean up a bit.

Reply:
I feel like it's fine for now. In my renderer rework I'll probably go ahead and do that though.